### ISSUE TITLE
Khan Academy extractor not working

### ISSUE THREAD

**Issue by TheCoderProYT** [yt-dlp/yt-dlp#3773]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.04.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [x] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Australia

### Description

Downloading of videos from Khan Academy, it says "400 Bad Request"

### Verbose log

```shell
[debug] Command-line config: ['-vU', 'https://www.khanacademy.org/computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8 (No ANSI), err utf-8 (No ANSI), pref UTF-8
[debug] yt-dlp version 2022.04.08 [7884ade65]
[debug] Python version 3.10.2 (CPython 64bit) - macOS-12.3-x86_64-i386-64bit
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1, rtmpdump 2.4
[debug] Optional libraries: brotli, certifi, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2022.04.08, Current version: 2022.04.08
yt-dlp is up to date (2022.04.08)
[debug] [khanacademy] Extracting URL: https://www.khanacademy.org/computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9
[khanacademy] computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9: Downloading JSON metadata
ERROR: [khanacademy] computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9: Unable to download JSON metadata: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 641, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/khanacademy.py", line 30, in _real_extract
    component_props = self._parse_json(self._download_json(
  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1029, in _download_json
    res = self._download_json_handle(
  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1008, in _download_json_handle
    res = self._download_webpage_handle(
  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 800, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 785, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 767, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/Cellar/yt-dlp/2022.4.8/libexec/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3601, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
```

**Comment by HobbyistDev** [comment]
It's work at version 2022.05.18

Here's the log :
```
C:\Users\USER\Downloads>yt-dlp -v -s "https://www.khanacademy.org/computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9"
[debug] Command-line config: ['-v', '-s', 'https://www.khanacademy.org/computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9']
[debug] Portable config "C:\Users\USER\Downloads\yt-dlp\yt-dlp.conf": ['-v', '--netrc-location', 'C:UsersUSERDownloadsyt-dlp.netrc', '-f', 'bv+ba/b', '--compat-options', 'format-sort', '--embed-metadata', '--embed-subs', '--sub-langs', 'all', '--console-title', '--downloader', 'aria2c']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d523] (win_exe)
[debug] Compatibility options: format-sort
[debug] Python version 3.8.10 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] exe versions: ffmpeg 5.0.1-full_build-www.gyan.dev (setts), ffprobe 5.0.1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] [khanacademy] Extracting URL: https://www.khanacademy.org/computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9
[khanacademy] computing/computer-science/informationtheory/info-theory/v/history-of-the-alphabet-language-of-coins-3-9: Downloading JSON metadata
[debug] [youtube] Extracting URL: 6NrTrBzC6dk
[youtube] 6NrTrBzC6dk: Downloading webpage
[youtube] 6NrTrBzC6dk: Downloading android player API JSON
[debug] Sort order given by user: hasaud, lang, quality, tbr, filesize, vbr, height, width, proto, vext, abr, aext, fps, fs_approx, source, id
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, hasaud, lang, quality, tbr, filesize, vbr, height, width, proto, vext, abr, aext, fps, fs_approx, source, id, res, hdr:12(7), vcodec:vp9.2(10), acodec, asr
[debug] Downloading subtitles: bn, bg, cs, en, it, ja, ko, pl, pt-BR, ro, es-419, th, vi
[info] 6NrTrBzC6dk: Downloading 1 format(s): 136+140
```

**Comment by pukkandan** [comment]
Fixed in 4f7a98c565873ea7a758efcd86e4296b6a06e817
